files/common/lib/systemd/system/delphix-migration.service

```diff
@@ -52,4 +55,4 @@ RemainAfterExit=yes
 ExecStartPost=/bin/systemctl disable delphix-migration.service
 
 [Install]
-WantedBy=default.target
+RequiredBy=delphix.target
```

Comment:
So from my understanding, this means that whenever `delphix-migration.service` is enabled, this will add an effective `Requires=delphix-migration.service` dependency to `delphix.target`. Whenever `delphix-migration.service` is disabled, it won't be there, right?

Reply:
Correct. For example:
```
$ sudo systemctl is-enabled delphix-migration
disabled
$ systemctl show -p Requires delphix.target
Requires=delphix-platform.service
```
```
$ sudo systemctl enable delphix-migration
Created symlink /etc/systemd/system/delphix.target.requires/delphix-migration.service → /lib/systemd/system/delphix-migration.service.
$ sudo systemctl is-enabled delphix-migration
enabled
$ systemctl show -p Requires delphix.target
Requires=delphix-platform.service delphix-migration.service
```

Reply:
Actually, I've experimented a bit more. The "restart" functionality is not as important for migration, I think having `RequiredBy=delphix-platform.service` and `RequiredBy=rtslib-fb-targetctl.service` would be very useful though, since that would prevent those services from failing when migration fails. I'll still add my pre-exec check for `/var/delphix/migration/perform-migration`, but it should not be triggered anymore unless there is a failure. This would also mean that doing a `systemctl start/restart delphix-mgmt` after migration has failed would effectively re-try the migration, since delphix-migration will be in its Required chain.